pkg/generic/descriptor_provider.go

```diff
@@ -24,3 +24,13 @@ type DescriptorProvider interface {
 	// Provide return a channel for provide service descriptors
 	Provide() <-chan *descriptor.ServiceDescriptor
 }
+
+// GetProviderOption provide options for descriptor provider
+type GetProviderOption interface {
+	Option() ProviderOption
+}
+
+type ProviderOption struct {
+	// DynamicGoExpected is if dynamicgo is expected to be used
+	DynamicGoExpected bool
```

Comment:
I thought the conclusion is to keep onlyy `DynamicGoEnabled`, is there any update?

Reply:
@felix021 Oh I should just rename it.

Reply:
done